src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -320,19 +320,12 @@ internal void UpdateAuthToken(TokenRefreshedParams tokenRefreshedParams)
         public ConnectionCompleteParams ValidateConnectParams(ConnectParams connectionParams)
         {
             string paramValidationErrorMessage;
-            if (connectionParams == null)

```

Comment:
why removing the null check here? is it never going to be null?

Reply:
That is correct, according to the test code, it appears to be a non-nullable type when I attempted to add a null test.

Reply:
If you're talking about the green squigglies (and compiler warnings) you should be aware that those are purely from a null-checking perspective. I.e. the fact that it's declared as a non-nullable parameter means that the compiler assumes that it's not being passed a null value. 

BUT this is not (yet) guaranteed to actually be the case. This is only enabled as a warning for now because the code base is so massive that trying to fix all these would be a huge task. The warnings are there mostly as a guide for when you're writing new code you use the nullable types correctly (like in typescript with strict mode on). 

In this specific case though I think it's probably fine. I would dearly hope that we never get passed null parameters (and if we did it should be an error farther up the chain) so removing the check here is fine by me.